plan/planbuilder.go

```diff
@@ -553,13 +557,23 @@ func (b *planBuilder) buildShow(show *ast.ShowStmt) Plan {
 	case ast.ShowWarnings:
 		p.SetSchema(buildShowWarningsSchema())
 	default:
+		switch showTp {
+		case ast.ShowTables, ast.ShowTableStatus:
+			if p.DBName == "" {
+				b.err = ErrNoDB
+				return nil
+			}
+		}
 		p.SetSchema(buildShowSchema(show))
 	}
 	for i, col := range p.schema.Columns {
 		col.Position = i
 	}
 	var conditions []expression.Expression
 	if show.Pattern != nil {
+		show.Pattern.Expr = &ast.ColumnNameExpr{
```

Comment:
Comment this. Should check show.Pattern.Expr == nil ?

Reply:
Noop, this is the first time that show.Pattern.Expr is assigned.